app/Controllers/configureController.php

```diff
@@ -353,27 +355,29 @@ public function queryAction(): void {
 		$query = new FreshRSS_UserQuery(FreshRSS_Context::$user_conf->queries[$id], $feed_dao, $category_dao, $tag_dao);
 		$this->view->query = $query;
 		$this->view->queryId = $id;
-		$this->view->categories = $category_dao->listCategories(false);
+		$this->view->categories = $category_dao->listCategories(false) ?: [];
 		$this->view->feeds = $feed_dao->listFeeds();
-		$this->view->tags = $tag_dao->listTags();
+		$this->view->tags = $tag_dao->listTags() ?: [];
 
 		if (Minz_Request::isPost()) {
+			/** @var array<string,string|array<string,string>> $params */
 			$params = array_filter(Minz_Request::paramArray('query'));
-			if (!empty($params['search'])) {
-				$params['search'] = htmlspecialchars_decode($params['search'], ENT_QUOTES);
+			$queryParams = [];
+			if (!empty($params['search']) && is_string($params['search'])) {
+				$queryParams['search'] = htmlspecialchars_decode($params['search'], ENT_QUOTES);
 			}
-			if (!empty($params['state'])) {
-				$params['state'] = array_sum($params['state']);
+			if (!empty($params['state']) && is_array($params['state'])) {
+				$queryParams['state'] = (int)(array_sum($params['state']));
 			}
-			$params['url'] = Minz_Url::display(['params' => $params]);
+			$queryParams['url'] = Minz_Url::display(['params' => $params]);
 			$name = Minz_Request::paramString('name') ?: _t('conf.query.number', $id + 1);
 			if ('' === $name) {
 				$name = _t('conf.query.number', $id + 1);
 			}
-			$params['name'] = $name;
+			$queryParams['name'] = $name;
 
 			$queries = FreshRSS_Context::$user_conf->queries;
-			$queries[$id] = new FreshRSS_UserQuery($params, $feed_dao, $category_dao, $tag_dao);
+			$queries[$id] = (new FreshRSS_UserQuery($queryParams, $feed_dao, $category_dao, $tag_dao))->toArray();
```

Comment:
One of the `->toArray()` that was called magically by `FreshRSS_ConfigurationSetter` removed in https://github.com/FreshRSS/FreshRSS/pull/5251 and broken since